File: frontend/src/types/job.ts
export interface Job {
  id: string;
  title: string;
  company: string;
  location: string;
  description: string;
  url: string;
  source: string;
  category?: string | null;
  job_type?: string | null;
  published_at?: string | null;
  created_at: string;
  updated_at: string;
  tags?: string[];
}

export interface JobListResponse {
  items: Job[];
  total: number;
  page: number;
  limit: number;
  pages: number;
}

export interface IngestionResult {
  source: string;
  fetched: number;
  inserted: number;
  updated: number;
  duplicates: number;
  failed: number;
  status: 'success' | 'partial_success' | 'failed';
  error_message?: string | null;
  duration_seconds: number;
}

export interface StatsResponse {
  total_jobs: number;
  sources_count: Record<string, number>;
  top_locations: Record<string, number>;
  categories_count: Record<string, number>;
  last_ingestion?: {
    source: string;
    status: string;
    fetched: number;
    inserted: number;
    updated: number;
    duplicates: number;
    completed_at?: string;
  } | null;
}

export interface AnalyticsResponse {
  total_jobs: number;
  tech_stack_distribution: Record<string, number>;
  sources_distribution: Record<string, number>;
  top_hiring_companies: Record<string, number>;
  locations_distribution: Record<string, number>;
}

export interface JobFilters {
  search: string;
  location: string;
  source: string;
  category: string;
  page: number;
  limit: number;
}

File: backend/app/schemas/job.py
from datetime import datetime
from typing import List, Optional, Dict
from pydantic import BaseModel, Field

class JobBase(BaseModel):
    title: str
    company: str
    location: str = "Remote"
    description: str
    url: str
    source: str
    category: Optional[str] = None
    job_type: Optional[str] = None
    published_at: Optional[datetime] = None

class JobSchema(JobBase):
    id: str
    created_at: datetime
    updated_at: datetime
    tags: List[str] = Field(default_factory=list)

    class Config:
        from_attributes = True

class JobListResponse(BaseModel):
    items: List[JobSchema]
    total: int
    page: int
    limit: int
    pages: int

class IngestionResult(BaseModel):
    source: str
    fetched: int
    inserted: int
    updated: int
    duplicates: int
    failed: int
    status: str
    error_message: Optional[str] = None
    duration_seconds: float

class StatsResponse(BaseModel):
    total_jobs: int
    sources_count: Dict[str, int]
    top_locations: Dict[str, int]
    categories_count: Dict[str, int]
    last_ingestion: Optional[dict] = None

class AnalyticsResponse(BaseModel):
    total_jobs: int
    tech_stack_distribution: Dict[str, int]
    sources_distribution: Dict[str, int]
    top_hiring_companies: Dict[str, int]
    locations_distribution: Dict[str, int]

class HealthResponse(BaseModel):
    status: str
    app_name: str
    version: str
    database: str
    timestamp: datetime
